language-design.md

```diff
@@ -27,7 +54,11 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
-
+One of the main themes of _Growing a Language_ is summarized by the quote at the bottom of page 12, "A good programmer builds a working vocabulary. In other words, a good programmer
+does language design, though not from scratch, but by building on the frame of a base
+language." [Steele, 1998]. This is what we did with our lab assignments, by building vocabulary for people to manipulate PCM audio. This was not just building vocabulary for use in the Python programming language, but also building vocabulary for use _with the rest of the library_. The original design of the lab assignment did not allow chaining multiple pieces of the vocabulary together, so any function needed to be a part of the library from the beginning (a cathedral-style design). We re-designed the library to allow the functions to be combined with each other, allowing a more bazaar-like approach. This is a better library design because "in the bazaar style of building a program or designing
+a language or what you will, the plan can change in real time to meet the needs of those
+who work on it and use it." [Steele, 1998].
```

Comment:
I don't know about you, but after I finished with the lab I was disappointed that CS5 had chosen to implement it the way they did - if they'd structured it as we did, the lab could've been a great introduction to classes! I think the lab was pretty early in the CS5 curriculum, though, so they may not have been ready for classes at that point.